### PR TITLE
Enhance the Go-Report card #130

### PR DESCRIPTION
**Pull request by fodil-a**
Fix of a big number of report-card issues (except .gitignore)

### CODE REVIEW

**Comment by fodil-a** [comment]
The check failed on uploading the codecov report, with this error:
```
Please provide an upload token from codecov.io with valid arguments
```

**Comment by fodil-a** [comment]
Fixes #130

**Comment by owulveryck** [comment]
This is weird because the token is in the secrets of github.
I will have a look at it.

Thanks for this PR.

**Comment by owulveryck** [comment]
I relaunched the job, and it is now failing on an initial step with the error 
`##[error]fatal: reference is not a tree: 1ca66614bc17265d1a55203fa9791006e470e9c1`

does this commit exist in your branch?

@diegobernardes @blackrez it looks like GitHub actions is not that magical after all :)))))))

File: backend/x/gorgonnx/softmax.go [hunk 1]
```diff
@@ -94,34 +94,31 @@ func coerceInto2D(n *gorgonia.Node, k int) (*gorgonia.Node, error) {
...
-		if err != nil {
-			return nil, err
-		}
-		a1, b1, err := gorgonia.Broadcast(n, m1, gorgonia.NewBroadcastPattern(nil, []byte{1}))
-		if err != nil {
-			return nil, err
-		}
-		return gorgonia.Sub(a1, b1)
-	default:
+	if len(n.Shape()) != 2 {
```
Comment (owulveryck) [inline backend/x/gorgonnx/softmax.go]: This is a lot clearer.

**Comment by fodil-a** [comment]
Thanks for the fast response.

Yes, it exists [here](https://github.com/fodil-a/onnx-go/commit/1ca66614bc17265d1a55203fa9791006e470e9c1)

**Comment by owulveryck** [comment]
I put Travis back (GitHub actions may need a bit more maturity).
Some builds are failing from times to time. I opened issue #138 to track the error which may not seem to be related to the code of this PR.

**Pull request merged**